### ISSUE TITLE
🙋‍♂️ Interactive Logger

### ISSUE THREAD

**Issue by davidnagli** [klauscfhq/signale#12]
It would be really cool if signale would provide a way to create interactive loggers (not sure if that's what there called, but lemme explain 😄)

Pretty much if I have an interactive CLI that continuously changes it's output, I want to override any previous output whenever I log something new.

For example, this:
```javascript
signale.pending();

setTimeout(()=>{
  signale.success();
}, 500)
```

Should output:
```
☐  pending
```

and then 500ms later replace it with:

```
✔  success
```

Instead of just writing both and ending up with both:
```
☐  pending
✔  success
```
This oubviously shouldn't be the default, but it would be really cool if this would be a feature that could be enabled.

tldr; add option to override previous output

**Comment by callensm** [comment]
This should be easily accomplished by adding a configuration option that prefixes specified logs with the `\033c` character. Logging that to the console clears/resets the terminal and start from the beginning as you described.

I can work on an implementation and do a PR if you want @klauscfhq

**Comment by klauscfhq** [comment]
Nice suggestion!
The [`stream.clearLine()`](https://nodejs.org/api/readline.html#readline_readline_clearline_stream_dir), [`stream.cursorTo()`](https://nodejs.org/api/readline.html#readline_readline_cursorto_stream_x_y) & [`stream.moveCursor()`](https://nodejs.org/api/readline.html#readline_readline_movecursor_stream_dx_dy) methods might also be useful.
Help is super welcome! : )

**Comment by callensm** [comment]
@davidnagli is your assumption that these `pending` logs would be blocking to the rest of the logs, meaning that nothing else will be logged until the pending process is resolved?
